File: instedit/pas/update_4op_flag_marks.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure update_4op_flag_marks;

var
  temp: Byte;

begin
  For temp := 1 to PRED(255) do
    If NOT check_4op_flag(temp) then
      begin
        If (songdata.instr_names[temp][1] in _4op_flag_chars) then
          songdata.instr_names[temp][1] := ' ';
        If (songdata.instr_names[SUCC(temp)][1] in _4op_flag_chars) then
          songdata.instr_names[SUCC(temp)][1] := ' ';
      end;
  For temp := 1 to PRED(255) do
    If check_4op_flag(temp) then
      begin
        If (songdata.instr_names[temp][1] = ' ') then
          songdata.instr_names[temp][1] := _4op_flag_chr_beg;
        If (songdata.instr_names[SUCC(temp)][1] = ' ') then
          songdata.instr_names[SUCC(temp)][1] := _4op_flag_chr_end;
      end;
end;

File: instedit/update_4op_flag_marks.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void update_4op_flag_marks (void)
{
  for (uint_least8_t i = 0; i < 254; i++)
    if (!check_4op_flag (i + 1))
    {
      unsigned char *c;

      c = &(GetStr (songdata.instr_names[i])[0]);
      if ((*c == charmap.four_op_upper_half) || (*c == charmap.four_op_lower_half))
        *c = ' ';
      c = &(GetStr (songdata.instr_names[i + 1])[0]);
      if ((*c == charmap.four_op_upper_half) || (*c == charmap.four_op_lower_half))
        *c = ' ';
    }

  for (uint_least8_t i = 0; i < 254; i++)
    if (check_4op_flag (i + 1))
    {
      unsigned char *c;

      c = &(GetStr (songdata.instr_names[i])[0]);
      if (*c == ' ')
        *c = charmap.four_op_upper_half;
      c = &(GetStr (songdata.instr_names[i + 1])[0]);
      if (*c == ' ')
        *c = charmap.four_op_lower_half;
    }
}
